### DISCUSSION TITLE
Question about SaveShare

### DISCUSSION THREAD

**Discussion by twnops** [wolveix/satisfactory-server#335] [General]
Hey there! Thanks for the work in this repo. It's insane how much work you've done here.

I want to get around the bugs introduced in the headless, dedicated software and I'm interested in the SaveShare. However, I'm curious about a few things:

First I'm operating under the following assumptions:
1. I understand this is spinning up an SFTP server.
2. I haven't looked at the source but I'm assuming you're monitoring save files and uploading them to the SFTP server in an automated fashion.

However, with respect to using this as a way to "collaborate", I'm confused how the game client is "playing" *from* the SFTP server or how it's uploaded in the case of the host not running their game? Am I missing something? How is this supposed to be setup?

Apologies in advance as it's been a while since I've used Go and haven't even begun looking thru the source.

**Comment by wolveix** [comment]
Hey @tmikel! With dedicated servers in their current state (post 1.0), I've found them to be very reliable. Performance issues are due to poor single-core performance from the server's CPU, or low RAM allocation.

Regarding SaveShare, the intended deployment is that everyone you intend to play with has it running on their local system. It works similarly to games like Grounded; so if one person (the current host) goes offline, someone else can just resume the hsoting responsibility as the save will have synced to their local system.

You're meant to run it alongside a normal multiplayer session, if that wasn't clear. All it's doing is providing a convenient method for all of the clients to automatically download the save

**Reply by twnops** [reply]
@wolveix Thank you for the response. I am a bit disappointed that I didnt realize this without asking. 

Regarding the single core speeds you mentioned, do you know the threshold in which you notice performance issues or bugs? Im very excited that you mentioned release 1.0 addressed some of those issues so Im looking forward to spinning up a new dedicated server this week!

**Reply by twnops** [reply]
Specifically remembering issues with belts (albeit mostly visual iirc). Not a deal breaker but I remember getting annoyed when looking at parts moving on conveyors and trying to eyeball how well things were moving along.

**Reply by wolveix** [reply]
@tmikel no problem! From helping people since 1.0, you generally want a CPU with a single-core passmark of 3000+. Passmark isn't a very reliable metric most of the time; however, it's a fairly good indicator for Satisfactory.

I personally host the game on a Ryzen 5900X, with 16GB RAM allocated to the game. Conveyors are much better now, and there are configurable settings in-game to adjust how far they're rendered etc. :)
